posts/postsRouter.js

```diff
@@ -0,0 +1,152 @@
+const express = require("express");
+const router = express.Router();
+const postDB = require("../data/helpers/postDb");
+const userDB = require("../data/helpers/userDb");
+
+// get post by id
+router.get("/:userID/posts/:postID", (req, res) => {
+  const { userID, postID } = req.params;
+  userDB
+    .getUserPosts(userID)
+    .then(posts => {
+      if (posts.length > 0) {
+        let user = posts[0].postedBy; // check for post id ownership, post id will otherwise show for incorrect user
+        postDB
+          .get(postID)
+          .then(post => {
+            if (post.postedBy === user) {
+              res.status(200).json(post);
+            } else {
+              res.status(404).json({
+                message: `Post ID ${postID} does not exist for ${user}.`
+              });
+            }
+          })
+          .catch(() =>
+            res
+              .status(500)
+              .json({ error: "The User information could not be retrieved." })
+          );
+      } else {
+        res
+          .status(404)
+          .json({ message: `User ID ${userID} does not have any Posts.` });
+      }
+    })
+    .catch(() =>
+      res
+        .status(500)
+        .json({ error: "The User information could not be retrieved." })
```

Comment:
You have this same callback elsewhere (with the same error text); I think it certainly calls for breaking it out into a helper function; for example:
```javascript
const sendServErr = res => () => (
  res
    .status(500)
    .json({ error: "The User information could not be retrieved." })
);
// ... in a route:
  .catch(sendServErr(res))
```
Eventually we'll see this broken out into another piece of middleware, but minimizing the size of repetition like this this is a large improvement!